hw_3/Finder.js

```diff
@@ -0,0 +1,227 @@
+const EventEmitter = require('events');
+const { readdir } = require('fs');
+const { extname } = require('path');
+const path = require("path");
+const chalk = require("chalk");
+
+const { createReadStream, createWriteStream } = require("fs");
+const FileType = require('file-type');
+// const { isExtMatched } = require('./functions/functions.js');
+
+class Finder extends EventEmitter {
+    constructor(deep, fileParams) {
+        super();
+
+        this._deep = deep;
+        this._path = fileParams.currPath;
+        this._ext = fileParams.ext;
+        this._search = fileParams.search;
+        this._regExp = new RegExp(fileParams.regExp);
+        this._dirProcessCount = 0;
+        this._fileProcessCount = 0;
+
+        this._logFileName;
+
+        this._interval = 2000;
+        this._timer;
+
+        this.on('file', this._setTimer);
+
+        setTimeout(() => {
+            this.emit('started');
+        }, 0);
+    }
+
+    async parse() {
+        this._logFileName = new Date(Date.now()).toLocaleString("ru-RU", {hour12: false}).replace(",", "").replace(/:/g, "-");
+
+        this._setTimer();
+        this._deep = (this._deep === 0) ? -1 : this._deep;
+
+        const result = await this._parseDir(this._path, this._deep);
+
+        this.emit('finished', result);
+        clearTimeout(this._timer);
+    }
+
+    async _parseDir(currPath, deep) {
+        let result = [];
+        let items;
```

Comment:
```suggestion
        let items = [];
```
иначе, в случае ошибки при readdir - `Uncaught TypeError: undefined is not iterable`